src/strings.ts

```diff
@@ -0,0 +1,25 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+'use strict';
+
+const _formatRegexp = /{(\d+)}/g;
+
+/**
+ * Helper to produce a string with a variable number of arguments. Insert variable segments
+ * into the string using the {n} notation where N is the index of the argument following the string.
+ * @param value string to which formatting is applied
+ * @param args replacements for {n}-entries
+ */
+export function format(value: string, ...args: any[]): string {
```

Comment:
How do we plan to use this? It's not that easy to implement it. For instance, ```Hello {{0}} world!``` should produce ```Hello {0} world!``` string, otherwise there is no way to keep ```{n}``` in the string.
Maybe there is a library that we can use that has all the corner-cases covered?

Reply:
There is no library that does this. What I noticed in ADS is that they use nls.localize and that takes arguments. Maybe we can do the same.

Reply:
Yeah, for all strings that are displayed to the user we need to use ```localize``` anyway. If that' the only place we need it - then it should not be a problem.